Error: The terraform-provider-postgresql_v1.14.0 plugin crashed!

Hello,
Facing the following problem whet trying to deploy 2 databases on 1 RDS database instance

### Terraform Version
Terraform v1.1.6
on darwin_arm64
+ provider registry.terraform.io/cyrilgdn/postgresql v1.14.0
+ provider registry.terraform.io/hashicorp/aws v3.74.3
+ provider registry.terraform.io/hashicorp/random v3.1.0

### Affected Resource(s)
couldn't create a database

### Terraform Configuration Files
```


#Define secret manager key

resource "random_password" "password" {
  length  = 32
  special = false
}

resource "aws_secretsmanager_secret" "staging-db" {
  name = "odys-pg-staging"
}

resource "aws_secretsmanager_secret_version" "sversion" {
  secret_id     = aws_secretsmanager_secret.secret-staging-db.id
  secret_string = <<EOF
   {
    "username": "odys",
    "password": "${random_password.password.result}"
   }
EOF
}

# Defining RDS Postgres Instance

resource "aws_db_instance" "staging-db" {
  allocated_storage       = 5 # gigabytes
  backup_retention_period = 7 # in days
  engine                      = "postgres"
  engine_version              = "13.5"
  identifier                  = "db-staging"
  instance_class              = "db.t3.micro"
  multi_az                    = false
  name                        = var.name
  password                    = local.db_creds.password
  port                        = 5432
  publicly_accessible         = true
  storage_encrypted           = true
  storage_type                = "gp2"
  username                    = local.db_creds.username
  allow_major_version_upgrade = true
  skip_final_snapshot         = true
  apply_immediately           = true
}
provider "postgresql" {
  scheme   = "awspostgres"
  host     = "db-host.rds.amazonaws.com"
  port     = 5432
  username = local.db_creds.username
  password = local.db_creds.password
  expected_version  = "13.5"
}
resource "postgresql_database" "staging-db" {
  name  = "sample_db"
  owner = "sample_owner"
}
```
```

### Panic Output
Error: Plugin did not respond
│ 
│   with postgresql_database.odys-staging-nt,
│   on odys-postgres.tf line 129, in resource "postgresql_database" "odys-staging-nt":
│  129: resource "postgresql_database" "odys-staging-nt" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-postgresql_v1.14.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104d800f4]

goroutine 15 [running]:
database/sql.(*Tx).grabConn(0x0, 0x105247340, 0x14000122008, 0x10467f998, 0x1057c6e20, 0x14000216500, 0x2)
        database/sql/sql.go:2112 +0x44
database/sql.(*Tx).ExecContext(0x0, 0x105247340, 0x14000122008, 0x104e502aa, 0x19, 0x0, 0x0, 0x0, 0x88, 0x140002900d0, ...)
        database/sql/sql.go:2362 +0x38
database/sql.(*Tx).Exec(...)
        database/sql/sql.go:2372
github.com/terraform-providers/terraform-provider-postgresql/postgresql.pgLockRole(0x0, 0x140000341c8, 0x4, 0x0, 0x105231d38)
        github.com/terraform-providers/terraform-provider-postgresql/postgresql/helpers.go:461 +0x60
github.com/terraform-providers/terraform-provider-postgresql/postgresql.createDatabase(0x1400007d7a0, 0x1400048a230, 0x0, 0x0)
        github.com/terraform-providers/terraform-provider-postgresql/postgresql/resource_postgresql_database.go:128 +0x1098
github.com/terraform-providers/terraform-provider-postgresql/postgresql.resourcePostgreSQLDatabaseCreate(0x1400007d7a0, 0x1400048a230, 0x0, 0x0)
        github.com/terraform-providers/terraform-provider-postgresql/postgresql/resource_postgresql_database.go:110 +0x30
github.com/terraform-providers/terraform-provider-postgresql/postgresql.PGResourceFunc.func1(0x1400048a230, 0x1050cc9e0, 0x1400000ac60, 0x2, 0x10580f2e0)
        github.com/terraform-providers/terraform-provider-postgresql/postgresql/helpers.go:22 +0x84
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0x1400009e580, 0x14000286050, 0x14000558f60, 0x1050cc9e0, 0x1400000ac60, 0x1050df901, 0x140007c77b8, 0x14000282630)
        github.com/hashicorp/terraform-plugin-sdk@v1.0.0/helper/schema/resource.go:305 +0x294
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0x1400009ea80, 0x140004aba20, 0x14000286050, 0x14000558f60, 0x140007cbee8, 0x1400012ecd8, 0x1050e18c0)
        github.com/hashicorp/terraform-plugin-sdk@v1.0.0/helper/schema/provider.go:289 +0x88
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0x140000927d0, 0x1052473b0, 0x140001157d0, 0x1400007d380, 0x140000927d0, 0x1049a78c0, 0x105193f00)
        github.com/hashicorp/terraform-plugin-sdk@v1.0.0/internal/helper/plugin/grpc_provider.go:885 +0x6b0
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1051dd140, 0x140000927d0, 0x1052473b0, 0x140001157d0, 0x1400007d320, 0x0, 0x1052473b0, 0x140001157d0, 0x14000154180, 0x174)
        github.com/hashicorp/terraform-plugin-sdk@v1.0.0/internal/tfplugin5/tfplugin5.pb.go:3189 +0x1c8
google.golang.org/grpc.(*Server).processUnaryRPC(0x140007a8540, 0x1052523f8, 0x14000083680, 0x1400084ff00, 0x140004a25d0, 0x1057d0060, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.34.0/server.go:1210 +0x3e8
google.golang.org/grpc.(*Server).handleStream(0x140007a8540, 0x1052523f8, 0x14000083680, 0x1400084ff00, 0x0)
        google.golang.org/grpc@v1.34.0/server.go:1533 +0xa50
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400083a230, 0x140007a8540, 0x1052523f8, 0x14000083680, 0x1400084ff00)
        google.golang.org/grpc@v1.34.0/server.go:871 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.34.0/server.go:869 +0x1f8

Error: The terraform-provider-postgresql_v1.14.0 plugin crashed!

### Expected Behavior
create the db/or at least report another understandable error 

### Actual Behavior
the plugin crashed

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

--- a/postgresql/resource_postgresql_database.go
+++ b/postgresql/resource_postgresql_database.go
@@ -125,6 +125,9 @@ func createDatabase(db *DBConnection, d *schema.ResourceData) error {
 		// Take a lock on db currentUser to avoid multiple database creation at the same time
 		// It can fail if they grant the same owner to current at the same time as it's not done in transaction.
 		lockTxn, err := startTransaction(db.client, "")
+		if err != nil {
+			return err
+		}
 		if err := pgLockRole(lockTxn, currentUser); err != nil {
 			return err
 		}

